pkg/util/grpcutil/grpc_util.go

```diff
@@ -29,6 +29,19 @@ import (
 	"google.golang.org/grpc/transport"
 )
 
+type localRequestKey struct{}
```

Comment:
it's not technically grpc, but we don't have a good context_util, just stuff in `util/{log,tracing}`